srt-tokio/src/socket.rs

```diff
@@ -201,6 +222,14 @@ impl SrtSocket {
     pub fn settings(&self) -> &ConnectionSettings {
         &self.settings
     }
+
+    pub fn stats(&mut self) -> &mut impl Stream<Item = SocketStatistics> {
+        &mut self.statistics
+    }
+
+    pub fn statistics(&mut self) -> &mut impl Stream<Item = SocketStatistics> {
+        &mut self.statistics
+    }
```

Comment:
since the polling of statistics can interfere with the packet send rate, I decided to make the 

> Don't think we need two of these....
> 
> Anyways, I think this means that you can't get statistics when you use send_all, which is the most canonical way to use `SrtSocket`... (can't make to `&mut self` references)
> 
> I would probably prefer if there was a `pub fn statistics(&self) -> SrtSocketStatistics`, and that handle had a `read_and_clear(&self)`, `read(&self)` as well as a `clear_statistics(&self)` function, which allows you to poll statistics at any frequency, clear whenever you want, as well as atomically do a read+clear which is probably the most typical usecase.

I was imagining providing a function that splits the streams (input, output, statistics, and possibly even errors/events?), so that they can be read from separately, potentially even from separate tasks. This can address the `send_all` scenario you mention, but otherwise also lets you use `select!` and just call specific read methods on the socket directly.

I am not married to the current design. The main motivation I had was to avoid the possibility of polling at too high a frequency, which could impact throughput in high bandwidth scenarios.

Reply:
> I would probably prefer if there was a `pub fn statistics(&self) -> SrtSocketStatistics`, and that handle had a `read_and_clear(&self)`, `read(&self)` as well as a `clear_statistics(&self)` function, which allows you to poll statistics at any frequency, clear whenever you want, as well as atomically do a read+clear which is probably the most typical usecase.

Ok @russelltg with the last commit `statistics()` now returns a `&mut impl Stream<Item=SocketStatistics> + Clone`. If you want to access this stream from a separate task you can now `clone()` the statistics stream.

Reply:
That seems like a reasonable starting point--it's simple and operable. Further improvements to the API can come, maybe with some research on how other libraries implement it.